server/app/models/PartModel.ts

```diff
@@ -1,23 +1,26 @@
+import { partType } from '../entities/Part';
 import { Document, Schema } from 'mongoose';
 
 export interface IPart extends Document {
   name: string;
   quality: string;
   description: string;
-  type: string;
+  type: partType;
   color: string;
   finish: string;
   grade: string;
   detail: string;
+  stock: number;
 }
 
 export const PartSchema: Schema = new Schema({
   name: { type: Schema.Types.String, required: true, unique: true, immutable: true },
   quality: { type: Schema.Types.String, required: true },
   description: { type: Schema.Types.String, required: false },
-  type: { type: Schema.Types.String, required: true },
+  type: { type: Schema.Types.String, required: true, immutable: true },
   color: { type: Schema.Types.String, required: false },
   finish: { type: Schema.Types.String, required: false },
   grade: { type: Schema.Types.String, required: false },
   detail: { type: Schema.Types.String, required: false },
+  stock: { type: Schema.Types.Number, required: true },
```

Comment:
What happens if we `post` with a negative part stock amount?

Reply:
Entire app explodes and zeet kicks us off the platform

Reply:
that's what i thought

Reply:
kidding, validation happens at the controller level now. I was avoiding re writing validation in 2 places

Reply:
but we could have double validation here

Reply:
Ahhh I see it now, I thought I saw it. Didn't know we had 2 schema thingies